source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -262,18 +299,28 @@ static void *ns_dyn_mem_internal_alloc(const int16_t alloc_size, int direction)
 #endif
 }
 
+void *ns_mem_alloc(ns_mem_book_t *heap, ns_mem_block_size_t alloc_size)
+{
+    return ns_mem_internal_alloc(heap, alloc_size, -1);
+}
+
+void *ns_mem_temporary_alloc(ns_mem_book_t *heap, ns_mem_block_size_t alloc_size)
+{
+    return ns_mem_internal_alloc(heap, alloc_size, 1);
+}
+
 void *ns_dyn_mem_alloc(int16_t alloc_size)
 {
-    return ns_dyn_mem_internal_alloc(alloc_size, -1);
+    return ns_mem_alloc(static_context, (ns_mem_block_size_t)alloc_size);
```

Comment:
It would be better to change ns_dyn_mem functions to take the same ns_mem_block_size_t types, and drop the casts. (Not a functional change, as they're currently the same types).